politeiawww/codetracker/github/api/commit.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+const (
+	apiCommitURL             = `https://api.github.com/repos/%s/%s/commits/%s`
+	apiPullRequestCommitsURL = `https://api.github.com/repos/%s/%s/pulls/%d/commits?per_page=250&page=%d&sort=updated&direction=desc`
+)
+
+// FetchCommit requests a given commit based on organization, repo and hash
+func (a *Client) FetchCommit(org, repo string, sha string) (*PullRequestCommit, error) {
+	url := fmt.Sprintf(apiCommitURL, org, repo, sha)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	body, err := a.sendGithubRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var commit PullRequestCommit
+	err = json.Unmarshal(body, &commit)
+	if err != nil {
+		return nil, err
+	}
+
+	return &commit, nil
+}
+
+// FetchPullRequestCommits requests all of the commit hashes under a given pull
+// request based on the organization, repo, and pull request number.
+func (a *Client) FetchPullRequestCommitSHAs(org, repo string, prNum int) ([]string, error) {
```

Comment:
I am seeing a repeat of functions that are directly called but have the potential for taking a very long time to execute. Is this ok or will it hold up other things? This has potential for deadlocks too. You should reason about this code and see if you may need to implement some channels and goroutines to execute these, potentially, very long functions.

Reply:
Well due to the rate limiting of github requests, everything is run synchronously.  These update/crawler passes only happen every month in the middle of the night, so speed is not very important in the case.